src/morphology.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+// Representation of 3-d embedded cable morphology, independent of other
+// cell information.
+
+#include <vector>
+
+namespace nest {
+namespace mc {
+
+struct section_point {
+    double x, y, z, r;  // [µm], r is radius.
+};
+
+enum class section_kind {
+    soma,
+    dendrite,
+    axon,
+    none
+};
+
+struct section_geometry {
```

Comment:
Yup. This functionality is currently being added to the `morphology` class. I would then like to keep the `segment_geometry` as a struct, with the intent that external consumers of the object look, but only touch if they really want to take responsibility for what breaks.